lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1866,6 +1880,61 @@ derivePublicKey ctx (ApiT wid) (ApiT role_) (ApiT ix) = do
+                    pure ( unAddress $
+                           CA.paymentAddress discriminant (spendingFromKey bytes)
+                         , EnterpriseDelegating )
+                (Just (Credential (Right (ApiScript (ApiT script)))), Nothing) ->
+                    pure ( unAddress $
+                           CA.paymentAddress discriminant (spendingFromScript script)
+                         , EnterpriseDelegating )
+                (Nothing, Just (Credential (Left (ApiPubKey bytes)))) -> do
+                    let (Right stakeAddr) =
+                            CA.stakeAddress discriminant (stakingFromKey bytes)
+                    pure ( unAddress stakeAddr, RewardAccount )
+                (Nothing, Just (Credential (Right (ApiScript (ApiT script))))) -> do
+                    let (Right stakeAddr) =
+                            CA.stakeAddress discriminant (stakingFromScript script)
+                    pure ( unAddress stakeAddr, RewardAccount )
+                (Just (Credential (Left (ApiPubKey bytes1))), Just (Credential (Left (ApiPubKey bytes2)))) ->
+                    pure ( unAddress $
+                           CA.delegationAddress discriminant (spendingFromKey bytes1) (stakingFromKey bytes2)
+                         , EnterpriseDelegating )
+                (Just (Credential (Left (ApiPubKey bytes))), Just (Credential (Right (ApiScript (ApiT script))))) ->
+                    pure ( unAddress $
+                           CA.delegationAddress discriminant (spendingFromKey bytes) (stakingFromScript script)
+                         , EnterpriseDelegating )
+                (Just (Credential (Right (ApiScript (ApiT script)))), Just (Credential (Left (ApiPubKey bytes)))) ->
+                    pure ( unAddress $
+                           CA.delegationAddress discriminant (spendingFromScript script) (stakingFromKey bytes)
+                         , EnterpriseDelegating )
+                (Just (Credential (Right (ApiScript (ApiT script1)))), Just (Credential (Right (ApiScript (ApiT script2)))) )->
+                    pure ( unAddress $
+                           CA.delegationAddress discriminant (spendingFromScript script1) (stakingFromScript script2)
+                         , EnterpriseDelegating )
+                (Nothing, Nothing) -> fail "At least one credential is required"
+    pure $ AnyAddress addr addrType (fromInteger netTag)
+  where
+      toXPub = fromJust . CA.xpubFromBytes . pubToXPub
+      pubToXPub bytes = BS.append bytes bytes
+      netTag = 1
+      spendingFromKey = CA.PaymentFromKey . CA.liftXPub . toXPub
+      spendingFromScript = CA.PaymentFromScript . CA.toScriptHash
+      stakingFromKey = CA.DelegationFromKey . CA.liftXPub . toXPub
+      stakingFromScript = CA.DelegationFromScript . CA.toScriptHash
+{--
+      netTag = case testEquality (typeRep @n) (typeRep @'Mainnet) of
+          Just{}  -> 1
+          Nothing -> 0
+--}
```

Comment:
This one is escaping me -> I want n -> Int ... for now I hacked it to be 1 (Mainnet) to carry on with integration tests

Reply:
Note that this should not be in `core`, because it only concerns cardano-node and really is cardano-node specific. Plus, you should be able to simply re-use: https://github.com/input-output-hk/cardano-wallet/blob/368b9fc27249e1faf6577c020297b74d4cf6f161/lib/shelley/src/Cardano/Wallet/Shelley.hs#L203-L218

Reply:
thanks for this hint. did it in https://github.com/input-output-hk/cardano-wallet/pull/2253/commits/3cf22f47f91eba47a1ebee825df32040dab8d1d7 also did some reshuffling of code connected with HasNetworkId to use it